api/scpca_portal/models/library.py

```diff
@@ -104,3 +109,33 @@ def get_formats_from_file_paths(cls, file_paths: List[Path]) -> List[str]:
         if any(path for path in file_paths if common.ANNDATA_EXT == path.suffix):
             formats.append(Library.FileFormats.ANN_DATA)
         return formats
+
+    @classmethod
+    def get_project_libraries_from_download_config(
+        cls, project, download_configuration: Dict
+    ) -> List[Self]:
+        if download_configuration not in common.VALID_PROJECT_DOWNLOAD_CONFIGURATIONS:
+            raise ValueError("Invalid download configuration passed. Unable to retrieve libraries.")
+
+        return [
+            library
+            for library in project.libraries.filter(modality=download_configuration["modality"])
+            if download_configuration["format"] in library.formats
+        ]
+
+    @classmethod
+    def get_sample_libraries_from_download_config(
+        cls, sample, download_configuration: Dict
+    ) -> List[Self]:
+        if download_configuration not in common.VALID_SAMPLE_DOWNLOAD_CONFIGURATIONS:
+            raise ValueError("Invalid download configuration passed. Unable to retrieve libraries.")
+
+        return [
+            library
+            for library in sample.libraries.filter(modality=download_configuration["modality"])
+            if download_configuration["format"] in library.formats
+        ]
```

Comment:
```suggestion
        return sample.libraries.filter(
            modality=download_configuration["modality"],
            format__contains=download_configuration["format"]
        )
```

https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/fields/#contains

You can update the others as well.